pkg/lock/lock.go

```diff
@@ -20,46 +20,39 @@ package lock
 
 import (
 	"os"
-	"sync"
+	"sync/atomic"
```

Comment:
I think it's better to use high level and more readable sync.mutex than low level atomic package.

https://golang.org/pkg/sync/atomic/
> Package atomic provides low-level atomic memory primitives useful for implementing synchronization algorithms.

> These functions require great care to be used correctly. Except for special, low-level applications, synchronization is better done with channels or the facilities of the sync package.